src/test/java/school/faang/user_service/service/SkillServiceTest.java

```diff
@@ -0,0 +1,174 @@
+package school.faang.user_service.service;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import school.faang.user_service.dto.skill.SkillCandidateDto;
+import school.faang.user_service.dto.skill.SkillCreateDto;
+import school.faang.user_service.dto.skill.SkillDto;
+import school.faang.user_service.entity.Skill;
+import school.faang.user_service.entity.User;
+import school.faang.user_service.entity.recommendation.Recommendation;
+import school.faang.user_service.entity.recommendation.SkillOffer;
+import school.faang.user_service.execption.DataValidationException;
+import school.faang.user_service.mapper.SkillMapper;
+import school.faang.user_service.repository.SkillRepository;
+import school.faang.user_service.repository.UserRepository;
+import school.faang.user_service.repository.recommendation.SkillOfferRepository;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@ExtendWith(MockitoExtension.class)
+public class SkillServiceTest {
+    private static final long USER_ID = 1L;
+    private static final long SKILL_ID = 1L;
+
+    @Mock
+    private SkillRepository skillRepository;
+
+    @Mock
+    private SkillOfferRepository skillOfferRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private SkillMapper skillMapper;
+
+    @InjectMocks
+    private SkillService skillService;
+
+    private SkillDto skillDto;
+    private SkillCreateDto skillCreateDto;
+    private User user;
+    private Skill skill;
+
+    @BeforeEach
+    public void beforeEach() {
+        skillCreateDto = new SkillCreateDto();
+        skillCreateDto.setTitle("Java");
+
+        skillDto = new SkillDto();
+        skillDto.setTitle("Java");
+
+        user = new User();
+        user.setId(USER_ID);
+        user.setUsername("Alex");
+
+        skill = new Skill();
+        skill.setId(SKILL_ID);
+        skill.setTitle("Java");
+    }
+
+    @Test
+    public void testExistingSkillCreate() {
+        Mockito.when(skillRepository.existsByTitle("Java")).thenReturn(true);
+
+        Assert.assertThrows(
+                DataValidationException.class,
+                () -> skillService.create(skillCreateDto)
+        );
+    }
+
+    @Test
+    public void testNotExistingSkillCreate() {
+        Mockito.when(skillRepository.existsByTitle("Java")).thenReturn(false);
+
+        Mockito.when(skillMapper.toEntity(skillCreateDto)).thenReturn(new Skill());
+        skillService.create(skillCreateDto);
+        Skill skill = skillMapper.toEntity(skillCreateDto);
+        Mockito.verify(skillRepository, Mockito.times(1)).save(skill);
```

Comment:
```suggestion
        Mockito.when(skillRepository.existsByTitle("Java")).thenReturn(false);

        SkillDto dto = skillService.create(skillCreateDto);
        ArgumentCaptor<Skill> argumentCaptor = ArgumentCaptor.forClass(Skill.class);
        Mockito.verify(skillRepository, Mockito.times(1)).save(argumentCaptor.capture());
        Skill capturedSkill = argumentCaptor.getValue();
        
        assertEquals(skillCreateDto.getTitle(), capturedSkill.getTitle());
```
Про ArgumentCaptor есть в лекции про тесты

Reply:
Это ошибка?

Reply:
Это более правильная проверка, которая была в лекции